api/src/main/java/org/openmrs/module/reporting/report/processor/HttpReportProcessor.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+
+package org.openmrs.module.reporting.report.processor;
+
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.module.reporting.report.Report;
+import org.springframework.stereotype.Component;
+
+/**
+* A ReportProcessor which sends the rendered report via http POST
+*/
+
+@Component
+public class HttpReportProcessor implements ReportProcessor {
+
+	protected Log log = LogFactory.getLog(this.getClass());
+	
+	public static final String CONNECTION_URL = "connectionUrl";
+	public static final String SUBJECT = "subject";
+	public static final String ADD_REPORT = "addReport";
+	
+	/**
+	 * @see ReportProcessor#getConfigurationPropertyNames()
+	 */
+	@Override
+	public List<String> getConfigurationPropertyNames() {
+		List<String> ret = new ArrayList<String>();
+		ret.add(CONNECTION_URL);
+		ret.add(SUBJECT);
+		ret.add(ADD_REPORT );
+		return ret;
+	}
+
+	/**
+	 * Performs some action on the given report
+	 * 
+	 * @param report the Report to process
+	 */
+	@Override
+	public void process(Report report, Properties configuration) {
+		try {
+			URL url = new URL(configuration.getProperty(CONNECTION_URL));
+			HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+			connection.setRequestMethod("POST");
+			String outPutContentType = report.getOutputContentType();
+			connection.setRequestProperty("Content-Type", "outPutContentType; charset=UTF-8");
```

Comment:
Typo in outPutContentType?

Reply:
@mseaton i don't understand clearly what you mean here..please elaborate